src/qthotkey/gui.cc

```diff
@@ -0,0 +1,329 @@
+
+#include <libaudcore/i18n.h>
+#include <libaudcore/preferences.h>
+
+#include <QtCore/QStringList>
+#include <QtGui/QKeyEvent>
+#include <QtGui/QMouseEvent>
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QStyle>
+#include <QtX11Extras/QX11Info>
+
+#include <X11/XKBlib.h>
+
+static void destroy_callback();
+static void ok_callback();
+
+static QPrefWidget *last_instance = nullptr;
+
+static const char *event_desc[EVENT_MAX] = {
+	[EVENT_PREV_TRACK] = N_("Previous track"),
+	[EVENT_PLAY] = N_("Play"),
+	[EVENT_PAUSE] = N_("Pause/Resume"),
+	[EVENT_STOP] = N_("Stop"),
+	[EVENT_NEXT_TRACK] = N_("Next track"),
+	[EVENT_FORWARD] = N_("Step forward"),
+	[EVENT_BACKWARD] = N_("Step backward"),
+	[EVENT_MUTE] = N_("Mute"),
+	[EVENT_VOL_UP] = N_("Volume up"),
+	[EVENT_VOL_DOWN] = N_("Volume down"),
+	[EVENT_JUMP_TO_FILE] = N_("Jump to file"),
+	[EVENT_TOGGLE_WIN] = N_("Toggle player window(s)"),
+	[EVENT_SHOW_AOSD] = N_("Show On-Screen-Display"),
+	[EVENT_TOGGLE_REPEAT] = N_("Toggle repeat"),
+	[EVENT_TOGGLE_SHUFFLE] = N_("Toggle shuffle"),
+	[EVENT_TOGGLE_STOP] = N_("Toggle stop after current"),
+	[EVENT_RAISE] = N_("Raise player window(s)")
+};
+
+class QLineKeyEdit: public QLineEdit
+{
+public:
+	explicit QLineKeyEdit(QWidget *parent, HotkeyConfiguration &hotkey)
+		: QLineEdit(parent),
+		m_hotkey(hotkey)
+	{
+		set_keytext(0, 0);
+	}
+
+	void set_keytext(int key, int mask)
+	{
+		QString text;
+
+		if ((key == 0) && (mask == 0))
+		{
+			text = QString::fromLocal8Bit(_("(none)"));
+		}
+		else
+		{
+			static const char *modifier_string[] = { "Control", "Shift", "Alt", "Mod2", "Mod3", "Super", "Mod5" };
+			static const unsigned int modifiers[] = { ControlMask, ShiftMask, Mod1Mask, Mod2Mask, Mod3Mask, Mod4Mask, Mod5Mask };
+
+			QStringList strings;
+
+			KeySym keysym;
+			keysym = XkbKeycodeToKeysym(QX11Info::display(), key, 0, 0);
+			if (keysym == 0 || keysym == NoSymbol)
+			{
+				text = QString::fromLocal8Bit("#%1").arg(key);
+			}
+			else
+			{
+				text = QString::fromLocal8Bit(XKeysymToString(keysym));
+			}
+
+			for (int j = 0; j < sizeof(modifiers)/sizeof(modifiers[0]); ++j)
+			{
+				if (mask & modifiers[j])
+				{
+					strings.push_back(QString::fromLatin1(modifier_string[j]));
+				}
+			}
+
+			if (key != 0)
+			{
+				strings.push_back(text);
+			}
+
+			text = strings.join(QLatin1String(" + "));
+		}
+
+		setText(text);
+
+		m_hotkey.key = key;
+		m_hotkey.mask = mask;
+	}
+
+protected:
+	void keyPressEvent(QKeyEvent *event) override
+	{
+		set_keytext(event->nativeScanCode(), event->nativeModifiers());
+	}
+
+private:
+	HotkeyConfiguration &m_hotkey;
+};
+
+KeyControls::~KeyControls()
+{
+	delete combobox;
+	delete keytext;
+	delete button;
+}
+
+QPrefWidget::QPrefWidget(QWidget *parent)
+	: QWidget(parent),
+	main_widget_layout(new QVBoxLayout(this)),
+	information_pixmap(new QLabel(this)),
+	information_label(new QLabel(QString::fromLocal8Bit(_("Press a key combination inside a text field.")), this)),
```

Comment:
`QString::fromLocal8Bit()` is not necessary here, but you should use `audqt::translate_str()` to properly translate mnemonics to Qt style from GTK style, so that new strings are not added unintentionally.

Reply:
Replaced all _(...) calls with audqt::translate_str(...) calls. Kept N_(...) calls as they are.

There should be no new translatable strings except for one in the quoted code.
But this string change is intentional: since I dropped mouse buttons support, I also removed text mentioning this feature.